src/fork-stream.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import {PassThrough, Readable, Transform} from 'stream';
+
+export function forkStream(stream: NodeJS.ReadableStream) {
+  return new ForkedVinylStream(stream);
+}
+
+/**
+ * Forks a stream of Vinyl files, cloning each file before emitting on the fork.
+ */
+export class ForkedVinylStream extends Readable {
```

Comment:
Did you consider extending stream.Transform instead? That way most of the piping of input to output is handled automagically. It might be as simple as:

```javascript

import * as stream from 'stream';

export function forkStream(input: NodeJS.ReadableStream): NodeJS.WritableStream {
  const forkedStream = new ForkedVinylStream();
  forkedStream.pipe(input);
  return forkedStream;
}

class ForkedVinylStream extends stream.Transform {
  constructor() {
    super({objectMode: true});
  }
  _transform(file) {
    this.push(file.clone({deep: true, contents: true}));
  }
}
```